core/matrix/dense.cpp

```diff
@@ -506,6 +530,17 @@ void Dense<ValueType>::compute_squared_norm2_impl(LinOp* result) const
 }
 
 
+template <typename ValueType>
+void Dense<ValueType>::compute_mean_impl(LinOp* result) const
```

Comment:
either this should call `compute_mean(ptr_param<LinOp>, array` directly, or it should also get the `array` parameter and get called by `compute_mean`.